src/Umbraco.Web.UI.Client/src/common/directives/validation/nodirtycheck.directive.js

```diff
@@ -10,13 +10,12 @@ function noDirtyCheck() {
         require: 'ngModel',
         link: function (scope, elm, attrs, ctrl) {
 
-            var alwaysFalse = {
-                    get: function () { return false; },
-                    set: function () { }
-                };
-            Object.defineProperty(ctrl, '$pristine', alwaysFalse);
-            Object.defineProperty(ctrl, '$dirty', alwaysFalse);
+            // If no attribute value is "false", then skip and use default behaviour.
+            var dirtyCheck = scope.$eval(attrs.noDirtyCheck) === false;
+            if (dirtyCheck)
+                return;
 
+            ctrl.$setDirty = Utilities.noop;
```

Comment:
While I'm sure this works, I'm a tad worried it's too good to be true. The quite intricate solution with the `alwaysFalse` binding to `$pristine` and `$dirty` must be there for a reason... this is just a hunch, but I'm afraid your solution over simplifies things. Wouldn't it be prudent to keep the old solution here, just after the conditional dirty check handling?

Reply:
Not sure if there is a specific reason it change `$pristine`. In other alternatives of `no-dirty-check` directives I have seen in only change `$dirty`. Regarding the "Unsaved changes" dialog it only rely on `$dirty` property as far as I know.

Reply:
Yep I agree with you. As I tried to explain, it _should_ work fine, but the current solution seems quite explicitly done as-is. There can be two reasons for this: Either the current solution is over complicated and implemented as such due to lack of knowledge/understanding, or else the current solution covers edge cases we can't right now. My point is it can't hurt to retain the current solution from a better-safe-than-sorry standpoint?

Reply:
Yeah, not sure if `$pristine` has any effect on the "Unsaved changes" dialog though. The reason we have the `no-dirty-check` directive is to prevent the "Unsaved changes" dialog to show in certain circumstances.

It seems it only rely on `$dirty` before opening the overlay here:
https://github.com/umbraco/Umbraco-CMS/blob/0140d906a5a66d3ba512332e845a64d36169f911/src/Umbraco.Web.UI.Client/src/common/directives/validation/valformmanager.directive.js#L175-L177